pkg/blobstore/cloud_blob_access.go

```diff
@@ -59,7 +55,7 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 	}
```

Comment:
Could you also call `b.Discard()` inside the error path above? Sorry; should have mentioned that earlier.